crates/xline/src/utils/trace.rs

```diff
@@ -57,11 +45,43 @@ pub fn init_subscriber(
     let jaeger_fmt_layer = tracing_subscriber::fmt::layer()
         .with_filter(tracing_subscriber::EnvFilter::from_default_env());
 
-    tracing_subscriber::registry()
-        .with(log_file_layer)
+    let prev_layers = tracing_subscriber::registry()
         .with(jaeger_fmt_layer)
         .with(jaeger_online_layer)
-        .with(jaeger_offline_layer)
-        .try_init()?;
-    Ok(guard)
+        .with(jaeger_offline_layer);
+
+    match *log_config.path() {
+        Some(ref file_path) => {
+            if file_path.to_string_lossy() == "/std" {
+                if *log_config.rotation() != default_rotation() {
+                    warn!(
+                        "The log is output to the terminal, so the rotation parameter is ignored."
+                    );
+                }
+                let stdout_layer = tracing_subscriber::fmt::layer()
+                    .event_format(format().compact())
+                    .with_writer(std::io::stderr)
+                    .with_ansi(false)
+                    .with_filter(*log_config.level());
+
+                prev_layers.with(stdout_layer).try_init()?;
+                return Ok(None);
+            }
+            let mut guard = None;
+            let log_file_layer = log_config.path().as_ref().map(|log_path| {
+                let file_appender = file_appender(*log_config.rotation(), log_path, name);
+                // `WorkerGuard` should be assigned in the `main` function or whatever the entrypoint of the program is.
+                let (non_blocking, guard_inner) = tracing_appender::non_blocking(file_appender);
+                guard = Some(guard_inner);
+                tracing_subscriber::fmt::layer()
+                    .event_format(format().compact())
+                    .with_writer(non_blocking)
+                    .with_ansi(false)
+                    .with_filter(*log_config.level())
+            });
+            prev_layers.with(log_file_layer).try_init()?;
+            Ok(guard)
+        }
+        None => unreachable!(),
```

Comment:
Please add some information in this unreachable arm

Reply:
Ok